cg/meta/deliver/delivery_api.py

```diff
@@ -0,0 +1,245 @@
+"""Module for deliveries of workflow files"""
+
+import logging
+from pathlib import Path
+from typing import Iterable
+
+from housekeeper.store.models import File, Version
+
+from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.constants import delivery as constants
+from cg.constants.constants import DataDelivery
+from cg.exc import MissingFilesError
+from cg.meta.deliver.utils import (
+    create_link,
+    get_delivery_case_name,
+    get_delivery_dir_path,
+    get_case_tags_for_pipeline,
+    get_out_path,
+    get_sample_out_file_name,
+    get_sample_tags_for_pipeline,
+    include_file_case,
+    include_file_sample,
+)
+from cg.store import Store
+from cg.store.models import Case, CaseSample, Sample
+
+LOG = logging.getLogger(__name__)
+
+
+class DeliveryAPI:
+    """Deliver API for workflows files."""
+
+    def __init__(
+        self,
+        store: Store,
+        hk_api: HousekeeperAPI,
+        project_base_path: Path,
+        force_all: bool = False,
+        ignore_missing_bundles: bool = False,
+        dry_run: bool = False,
+    ):
+        self.store = store
+        self.hk_api = hk_api
+        self.project_base_path = project_base_path
+        self.ignore_missing_bundles = ignore_missing_bundles
+        self.deliver_failed_samples = force_all
+        self.dry_run = dry_run
+
+    def set_dry_run(self, dry_run: bool):
+        self.dry_run = dry_run
+
+    def set_deliver_failed_samples(self, deliver_failed_samples: bool):
+        self.deliver_failed_samples = deliver_failed_samples
+
+    def set_ignore_missing_bundles(self, ignore_missing_bundles: bool):
+        self.ignore_missing_bundles = ignore_missing_bundles
+
+    def deliver(self, ticket: str | None, case_id: str | None, pipeline: str) -> None:
```

Comment:
This sounds like a design problem. I think that this method should only accept a case id or ticket id by default. Then we retrieve the pipeline and data analysis from the case.

We can expose a separate method for the edge case where production wants to override the data analysis set for the case/deliver some specific analysis for the case.

Does that make sense?

Reply:
I like what you suggested! Since data delivery is set on case level, it shouldn't be required by default. I guess there might be reuploads and stuff done manually which can benefit from an override though.

Reply:
It very much does make sense. As you say we have all the information in the case and it would be nice to just aggregate the tags and call deliver once instead of looping over delivery_types.

Reply:
Follow up: can we simplify the logic in this module even further?

A delivery in the context of this module seems to mean: copy the files of a bundle version (the analysis) into the customers folder on Hasta. 

Do we really need to care about the analysis or data delivery at all to do this? Or is it because we need to filter the files stored in the bundle version?

Reply:
Leaving this for a separate PR